### ISSUE TITLE
Synchronization issues when having two GoogleMap instances on the same DOM

### ISSUE THREAD

**Issue by oscar-martin** [istarkov/google-map-react#35]
I haven't found any place in where I can know whether multiple GoogleMap components can be used inside the same DOM. So I gave it a try by putting two maps in different tabs and there are a couple of sync issues:

The two maps are centered in the same geo-position but both of them are located in different places in each tab with different frame sizes as well. 

Perceived issues on the second tab:

1.- the second map is not centered properly (the center seems to be a bit offset to the top-left corner.
2.- neighbours tiles from that center point do not appear in the DOM

If I expand a bit the browser window a bit then, neighbour tiles do appear. The offset thing is not fixed though.

Is it multiple map components in the same DOM supported?

**Comment by MozMorris** [comment]
Does [this example](http://jsfiddle.net/cvju3p2v/3/) help?

**Comment by oscar-martin** [comment]
Hi,

I do not manage to create the code in jsfiddle that resembles the problem but this is more related to my problem:


http://jsfiddle.net/cvju3p2v/19/ <http://jsfiddle.net/cvju3p2v/19/>


The problem seems to be related to the tab switches.

**Comment by istarkov** [comment]
IMHO Bootstrap tabs is broken,
This simple example is also not work properly:
http://jsfiddle.net/o958xp9u/

**Comment by jedwards1211** [comment]
I don't trust the `react-bootstrap` project either.  Dropdowns weren't going away when I clicked outside of them and there was an open issue about 'Dropdowns need an overhaul' with 10 linked issues that had been open for months last I checked.  I just wrote my own React components to use Bootstrap without jQuery.

**Comment by jedwards1211** [comment]
If you can reproduce the issue without `react-bootstrap

**Comment by oscar-martin** [comment]
Finally I managed to reproduce it:

http://jsfiddle.net/cvju3p2v/22/ 

That's exactly the issue I'm having.

**Comment by istarkov** [comment]
Mb you can implement this with React code?
I know your issue, please read this https://github.com/istarkov/google-map-react/issues/20.

**Comment by oscar-martin** [comment]
The problem I had with the suggested approach was not working as expected (maybe I did not take the proper amount of time to dig into it tough).

Finally I decided to use 'window.dispatchEvent(new Event('resize'))' during a callback I got when the second tab is clicked. Then the map gets updated and uncentered yet (so I do center it manually).

By the way, I needed to use setTimeout for calling "window.dispatchEvent(new Event('resize'))" because otherwise the updated did not occur. 

I'm starting with javascript so maybe I am doing something really bad here.
